src/BuiltInTools/dotnet-watch/HotReload/BlazorWebAssemblyDeltaApplier.cs

```diff
@@ -72,43 +72,43 @@ public async ValueTask ReportDiagnosticsAsync(DotNetWatchContext context, IEnume
         private async Task<bool> VerifyDeltaApplied(DotNetWatchContext context, CancellationToken cancellationToken)
         {
             var _receiveBuffer = new byte[1];
-            try 
+            try
             {
-                ValueWebSocketReceiveResult? result;
-                // Sometimes we can received a Close message from the WebSocket before the
-                // browser has transmitted the acknowledgement. To address this, we wait until
-                // a result has been received and it has the expected message type (binary).
-                // There is a `WaitAsync` at the invocation site of `VerifyDeltaApplied` so this
-                // will time out if we have received an appropriate message within 5 seconds.
-                do
+                // We want to give the client some time to ACK the deltas being applied. VerifyDeltaApplied is limited by a
+                // 5 second wait timeout enforced using a WaitAsync. However, WaitAsync only works reliably if the calling
+                // function is async. If BrowserRefreshServer.ReceiveAsync finishes synchronously, the WaitAsync would
+                // never have an opportunity to execute. Consequently, we'll give it some reasonable number of opportunities
+                // to loop before we decide that applying deltas failed.
+                for (var i = 0; i < 100; i++)
                 {
-                    result = await context.BrowserRefreshServer.ReceiveAsync(_receiveBuffer, cancellationToken);
-                } while (!result.HasValue || result?.MessageType is not WebSocketMessageType.Binary);
-                return IsDeltaApplied(result);
+                    var result = await context.BrowserRefreshServer.ReceiveAsync(_receiveBuffer, cancellationToken);
+                    if (result is null)
+                    {
+                        // A null result indicates no clients are connected. No deltas could have been applied in this state.
+                        _reporter.Verbose("No client is connected to ack deltas");
+                        return false;
+                    }
+
+                    if (IsDeltaApplied(result.Value))
```

Comment:
nit: return IsDeltaApplied

Reply:
It's inside a loop though. We want it to return true the first time it succeeds.

Reply:
Ah didn't notice it was in the loop